### ISSUE TITLE
Infinite loop in TableStepperBase on trySplit

### ISSUE THREAD

**Issue by lrytz** [scala/bug#11468]
This hangs
```
scala> collection.mutable.HashSet("1", "2").stepper.trySplit
```

here: https://github.com/scala/scala/blob/63769aff4d7301b595d5b3cd71a28c31313ff408/src/library/scala/collection/convert/impl/TableStepper.scala#L58

**Comment by lrytz** [comment]
cc @Ichoran

**Comment by lrytz** [comment]
(found while porting tests over from java8-compat, https://github.com/scala/scala/compare/2.13.x...lrytz:java8-tests?expand=1)

**Comment by SethTisue** [comment]
@lrytz is 2.13.0 a reasonable milestone for this?

**Comment by Ichoran** [comment]
I'll test the `trySplit`s tonight.  I think fixing them all will be practical in a short period of time.

**Comment by Ichoran** [comment]
Verified, and I think there are correctness issues with all the other set/map splitters.  The others all seem fine.  I'll investigate shortly.

**Comment by Ichoran** [comment]
`BitSet`s also keep returning empty steppers instead of eventually giving `null` on splits.  Bad practice, though the Java API tolerated it last I checked.

**Comment by lrytz** [comment]
I think we don't need to delay RC1 over these bugs, but should fix them for RC2 / .0

**Comment by Ichoran** [comment]
Yeah, okay.  I've fixed this one; will submit a PR to fix the TableStepper bug and the ChampStepper one (it skips elements).

**Comment by Ichoran** [comment]
Have a fix in https://github.com/Ichoran/scala/tree/issue-11468

Is 2.13.x still what I should rebase against?

**Comment by SethTisue** [comment]
yes

**Comment by lrytz** [comment]
@Ichoran I included your commit in my PR with the new tests: https://github.com/scala/scala/pull/7970. It fixes the issues I observed. If you have time to review this one, that would be great :)

**Comment by Ichoran** [comment]
Fix is part of #7970
